crates/subspace-malicious-operator/src/bin/subspace-malicious-operator.rs

```diff
@@ -0,0 +1,432 @@
+                    reserved_peers: cli.dsn_reserved_peers,
+                    // Override enabling private IPs with --dev
+                    allow_non_global_addresses_in_dht: cli.dsn_enable_private_ips
+                        || cli.run.shared_params.dev,
+                    max_in_connections: cli.dsn_in_connections,
+                    max_out_connections: cli.dsn_out_connections,
+                    max_pending_in_connections: cli.dsn_pending_in_connections,
+                    max_pending_out_connections: cli.dsn_pending_out_connections,
+                    external_addresses: cli.dsn_external_addresses,
+                    // Override initial Kademlia bootstrapping  with --dev
+                    disable_bootstrap_on_start: cli.dsn_disable_bootstrap_on_start
+                        || cli.run.shared_params.dev,
+                }
+            };
+
+            let consensus_chain_config = SubspaceConfiguration {
+                base: consensus_chain_config,
+                // Domain node needs slots notifications for bundle production.
+                force_new_slot_notifications: !cli.domain_args.is_empty(),
+                subspace_networking: SubspaceNetworking::Create { config: dsn_config },
+                sync_from_dsn: cli.sync_from_dsn,
+                enable_subspace_block_relay: cli.enable_subspace_block_relay,
+                // Timekeeper is enabled if `--dev` is used
+                is_timekeeper: cli.timekeeper || cli.run.shared_params.dev,
+                timekeeper_cpu_cores: cli.timekeeper_cpu_cores,
+            };
+
+            let partial_components = subspace_service::new_partial::<
+                PosTable,
+                RuntimeApi,
+                ExecutorDispatch,
+            >(
+                &consensus_chain_config.base, &pot_external_entropy
+            )
+            .map_err(|error| {
+                sc_service::Error::Other(format!("Failed to build a full subspace node: {error:?}"))
+            })?;
+
+            let keystore = partial_components.keystore_container.keystore();
+
+            let consensus_chain_node = subspace_service::new_full::<PosTable, _, _>(
+                consensus_chain_config,
+                partial_components,
+                true,
+                SlotProportion::new(3f32 / 4f32),
+            )
+            .await
+            .map_err(|error| {
+                sc_service::Error::Other(format!("Failed to build a full subspace node: {error:?}"))
+            })?;
+
+            (consensus_chain_node, keystore)
+        };
+
+        StorageMonitorService::try_spawn(
+            cli.storage_monitor,
+            database_source,
+            &consensus_chain_node.task_manager.spawn_essential_handle(),
+        )
+        .map_err(|error| {
+            sc_service::Error::Other(format!("Failed to start storage monitor: {error:?}"))
+        })?;
+
+        // Run a domain node.
+        if !cli.domain_args.is_empty() {
```

Comment:
this should be mandatory or else no point having such binary

Reply:
Right, will make it mandatory